source/dubregistry/repositories/github.d

```diff
@@ -44,28 +44,35 @@ class GithubRepository : Repository {
 
 	RefInfo[] getTags()
 	{
-		import std.datetime : SysTime;
-
-		Json tags;
-		try tags = readJson(getAPIURLPrefix()~"/repos/"~m_owner~"/"~m_project~"/tags?per_page=100");
-		catch( Exception e ) { throw new Exception("Failed to get tags: "~e.msg); }
+		import std.datetime.systime : SysTime;
+		import std.conv: text;
 		RefInfo[] ret;
-		foreach_reverse (tag; tags) {
-			try {
-				auto tagname = tag["name"].get!string;
-				Json commit = readJson(getAPIURLPrefix()~"/repos/"~m_owner~"/"~m_project~"/commits/"~tag["commit"]["sha"].get!string, true, true);
-				ret ~= RefInfo(tagname, tag["commit"]["sha"].get!string, SysTime.fromISOExtString(commit["commit"]["committer"]["date"].get!string));
-				logDebug("Found tag for %s/%s: %s", m_owner, m_project, tagname);
-			} catch( Exception e ){
-				throw new Exception("Failed to process tag "~tag["name"].get!string~": "~e.msg);
+		for (size_t page = 1; ; page++)
+		{
+			Json tags;
+			try tags = readJson(getAPIURLPrefix()~"/repos/"~m_owner~"/"~m_project~"/tags?per_page=100&page=" ~ page.text);
+			catch( Exception e ) { throw new Exception("Failed to get tags: "~e.msg); }
+			size_t count;
+			foreach_reverse (tag; tags) {
+				try {
+					count++;
+					auto tagname = tag["name"].get!string;
+					Json commit = readJson(getAPIURLPrefix()~"/repos/"~m_owner~"/"~m_project~"/commits/"~tag["commit"]["sha"].get!string, true, true);
+					ret ~= RefInfo(tagname, tag["commit"]["sha"].get!string, SysTime.fromISOExtString(commit["commit"]["committer"]["date"].get!string));
+					logDebug("Found tag for %s/%s: %s", m_owner, m_project, tagname);
+				} catch( Exception e ){
+					throw new Exception("Failed to process tag "~tag["name"].get!string~": "~e.msg);
+				}
 			}
+			if (count < 100)
+				break;
```

Comment:
Unfortunately that's not the right way to do pagination as you might send an unnecessary request if the count is 100.
You need to look at the "link" header.

```
$ curl -v "https://api.github.com/repos/dlang/dmd/tags?per_page=1" 
...
< link: <https://api.github.com/repositories/1257070/tags?per_page=1&page=2>; rel="next", <https://api.github.com/repositories/1257070/tags?per_page=1&page=258>; rel="last"
```

Reply:
This will affect less the %1 of repositories and with a probability less than %1. In total it affects less with the probability less than %0.01.

Reply:
Yes, but it will still result in an error at the dub registry.

Reply:
Why error?
https://api.github.com/repos/libmir/mir-algorithm/tags?per_page=100&page=10 returns an empty array and status 200.